test/regression/if.sol

```diff
@@ -0,0 +1,20 @@
+pragma solidity ^0.8.24;
+
+contract IfTest {
+
+    constructor() {
+      bool x = true;
+      if (x) {
+      } else {
+	require(false, "");
+      }
```

Comment:
My only comment is that when the if condition is false, both the if then and the full if then else are tested. Here, when the if condition is true, only the if then else case is tested. You may consider adding a check for the if then case.